samples/tracebld/trcbld.cpp

```diff
@@ -4496,11 +4496,15 @@ int WINAPI Mine_EntryPoint(VOID)
 
         FileNames::ParameterizeLine(pwzFin, pwzFin + wcNew);
         if (HasSpace(wzPath)) {
-            Tblog("<t:Line>&quot;%le&quot; %le</t:Line>\n", wzPath, pwzFin);
+            Tblog("<t:Line>&quot;%le&quot; ", wzPath);
         }
         else {
-            Tblog("<t:Line>%le %le</t:Line>\n", wzPath, pwzFin);
+            Tblog("<t:Line>%le ", wzPath);
         }
+        for(PWCHAR pwzTmp = pwzFin; pwzTmp < pwzFin + wcNew; pwzTmp += 32764) {
```

Comment:
if pwzTmp is very large, it overflows the structure that passes this string to the host app. This solution sends it in parts.

Reply:
I don't think this does what you think it does.   The problem is that the characters in the string are being escaped.  Some input characters require 6 characters in the output.  This means that the output may require more than 32764 characters.  if it does, some characters in the input string may not be written.

It is also not a good idea to embed a constant like 32764 here.  This constant comes from tracebld.h, specifically the number of characters in szMessage in the definition of the struct `_TBLOG_MESSAGE`.  In fact, looking at the code for VSafePrintf, it looks like one character is reserved for a null terminator (line 1880 in trcbld.cpp).